java-checks/src/main/java/org/sonar/java/checks/helpers/Javadoc.java

```diff
@@ -101,7 +101,7 @@ private static BlockTagKey of(BlockTag tag, @Nullable String name) {
   private static final Tree.Kind[] CLASS_KINDS = PublicApiChecker.classKinds();
   private static final Tree.Kind[] METHOD_KINDS = PublicApiChecker.methodKinds();
   private static final List<String> GENERIC_EXCEPTIONS = Arrays.asList("Exception", "java.lang.Exception");
-  private static final Pattern BLOCK_TAG_LINE_PATTERN = Pattern.compile("^@\\S+.*");
+  private static final Pattern BLOCK_TAG_LINE_PATTERN = Pattern.compile("^@.+");
```

Comment:
The regex itself is subject to catastrophic backtracking, when the input is like "@aaa\n", with thousands of "a". In practice, we already process the string that will be matched by the regex, so I don't think we are really subject to a problem here. Still, the regex can be simplified because there is no point in being that precise to detect "Block tag", a line starting with "@" is already enough.

Reply:
`@\\S.*` would be equivalent to the original without the backtracking issue. I don't know Javadoc well enough to know whether "the `@` sign must be followed by a non-space" is an important property to check or not. If not, you can probably get rid of the regex altogether and just use `.startsWith("@")` or `.charAt(0) == '@'`.